.github/workflows/format.yml

```
@@ -19,13 +19,13 @@ jobs:
         with:
           python-version: '3.11'
 
-      - name: Install dependencies
-        run:
-            python -m pip install click==8.0.4 black==23.7.0 isort==5.13.2
-
       - name: Checkout PennyLane-Lightning
         uses: actions/checkout@v3
 
+      - name: Install dependencies
+        run:
+            python -m pip install -r requirements-dev.txt
```

This will result the installation of many unused python packages. let's update this as follows,
 
```suggestion
            python -m pip install click==8.0.4 black==23.7.0 isort==5.13.2
```

---

Indeed, but dependabot won't update `format.yml` so since this is one of the quickest jobs I would pay the price of automation here.

---

Let's avoid splitting deps out of requirements-dev.txt. I'll resolve this

---

absolutely! in that case, let's fix them in all places is one single PR, not partially in multiple ones.